### PR TITLE
Fixed image file extension detection

### PR DESCRIPTION
**Pull request by MarioLiebisch**
* Previously this would fail on `jpeg` and would detect something such as `notajpg` as `jpg`.
* This fixes issue #929.

### CODE REVIEW

File: src/SFML/Graphics/ImageLoader.cpp [hunk 1]
```diff
@@ -227,35 +227,34 @@ bool ImageLoader::saveImageToFile(const std::string& filename, const std::vector
     if (!pixels.empty() && (size.x > 0) && (size.y > 0))
     {
         // Deduce the image type from its extension
-        if (filename.size() > 3)
+
+        // Extract the extension
+        const std::size_t dot = filename.find_last_of('.');
```
Comment (kimci86) [inline src/SFML/Graphics/ImageLoader.cpp]: To nitpick, I think that using `rfind` could be more appropriate.

File: src/SFML/Graphics/ImageLoader.cpp [hunk 2]
```diff
@@ -227,35 +227,34 @@ bool ImageLoader::saveImageToFile(const std::string& filename, const std::vector
...
+        const std::string extension = dot != std::string::npos ? toLower(filename.substr(dot + 1)) : "";
```
Comment (kimci86) [inline src/SFML/Graphics/ImageLoader.cpp]: I think the check for a valid `dot` index should be done first in an `if` embracing all the following code. What do you think ?
```c++
if (dot != std::string::npos)
{
    const std::string extension = toLower(filename.substr(dot + 1));
    if (extension == "bmp")
    //...
}
```

File: src/SFML/Graphics/ImageLoader.cpp (hunk 1 above)
Comment (LaurentGomila) [inline src/SFML/Graphics/ImageLoader.cpp]: > To nitpick, I think that using rfind could be more appropriate.

Why? For a single character it should make absolutely no difference.

File: src/SFML/Graphics/ImageLoader.cpp (hunk 2 above)
Comment (LaurentGomila) [inline src/SFML/Graphics/ImageLoader.cpp]: Let's not make optimization or special cases where we don't need to. "" is a valid extension, like any other.

File: src/SFML/Graphics/ImageLoader.cpp (hunk 1 above)
Comment (kimci86) [inline src/SFML/Graphics/ImageLoader.cpp]: In my opinion, it looks weird to use `find_last_of` with a single character. I think `rfind` make more sense. Anyway feel free to ignore this comment.

Comment (MarioLiebisch) [inline src/SFML/Graphics/ImageLoader.cpp]: Never really used `rfind()` (didn't actually know it :D). I guess for single characters the implementation is exactly the same. The only difference occurs when passing an actual string.

File: src/SFML/Graphics/ImageLoader.cpp (hunk 2 above)
Comment (MarioLiebisch) [inline src/SFML/Graphics/ImageLoader.cpp]: I thought about something similar as well, but I don't think it's worth making the code more elaborate. Also your snippet is excluding the case where there actually is a dot, but no extension (the file name could be `image.`). So you could actually check for an empty extension first to skip all further checks, but at the same time that's marginal complexity compared to everything else happening in that whole call.

**Comment by TankOs** [comment]
Looks good. :+1: :cat2:

**Comment by binary1248** [comment]
:+1:

**Comment by eXpl0it3r** [comment]
This PR has been added to my merge list, meaning it will be merged soon into the branch 2.3.x, unless someone raises any concerns.

**Comment by eXpl0it3r** [comment]
Merged in df99d5fee223d018e1a7a4c4a1f3372f76d4edf5 on branch 2.3.x

**Pull request closed without merging**